howso/direct/client.py

```diff
@@ -1565,7 +1588,7 @@ def remove_cases(
         # Convert session instance to id
         if (
             isinstance(condition, dict) and
-            isinstance(condition.get('.session'), Session)
+            isinstance(condition.get('.session'), Dict)
         ):
             condition['.session'] = condition['.session'].id
```

Comment:
It could be possible that the `.session` in the user specified condition is now a dict or engine.Session instance. This use to account for either the openapi.Session or engine.Session. In which case we may now want to support resolving the id in both cases. This also uses an incorrect accessor now, a dict will not have a `id` attribute here. I would suggest fixing the id accessor here for dict and do a similar check in engine.Trainee.remove_cases method to check if `.session` is an engine.Session and resolve the id.

Reply:
Addressed by: 5717250

Reply:
Continued fixes in: 646d3dd